psrqpy/search.py

```diff
@@ -1449,6 +1449,48 @@ def solfunc(m2, sini, mf, m1):
                     )
                 self.update(ECCERRnew, name='ECC_ERR')
 
+        # derive EPS1 and EPS2 from ECC and OM
+        reqpars = ['ECC', 'OM']
+        if np.all([p in self.columns for p in reqpars]):
+            ECC = self.catalogue['ECC'].values.copy()
+            OM = self.catalogue['OM'].values.copy()
+            EPS1new = np.full(self.catalogue_len, np.nan)
+            EPS2new = np.full(self.catalogue_len, np.nan)
+
+            idx = np.isfinite(ECC) & np.isfinite(OM)
+
+            EPS1new[idx] = ECC[idx] * np.sin(OM[idx])
```

Comment:
If you use `EPS1new = np.full(self.catalogue_len, np.nan)` and then do the calculation in this way, don't you overwrite the EPS1 (and EPS2) values that were previously in the catalogue but not associated with ECC/OM values (and vice versa with the previous calculations)? Or am I misunderstanding how `self.update` works? 

I would think you'd want to do: 
```if 'EPS1' in self.columns:
    EPS1new = self.catalogue['EPS1'].values.copy()
else:
    EPS1new = np.full(self.catalogue_len, np.nan)```
and so on for EPS2, and then for ECC and OM in the previous derivation.

Reply:
By default `self.update` is set to only update non-NaN values if they already exist in the table, so it shouldn't overwrite existing values.